Mathlib/Combinatorics/SimpleGraph/Tutte.lean

```diff
@@ -0,0 +1,125 @@
+/-
+Copyright (c) 2024 Pim Otte. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Pim Otte
+-/
+
+import Mathlib.Combinatorics.SimpleGraph.Matching
+import Mathlib.Combinatorics.SimpleGraph.Operations
+import Mathlib.Combinatorics.SimpleGraph.UniversalVerts
+import Mathlib.Data.Fintype.Card
+
+/-!
+# Tutte's theorem (work in progress)
+
+## Main definitions
+* `SimpleGraph.TutteViolator G u` is a set of vertices `u` which certifies non-existance of a
+  perfect matching.
+-/
+
+namespace SimpleGraph
+
+universe u
+variable {V : Type u} {G G' : SimpleGraph V} {u x v' w : V} [Fintype V]
+
+/-- A set certifying non-existence of a perfect matching. -/
+def IsTutteViolator (G : SimpleGraph V) (u : Set V) : Prop :=
+  u.ncard < ((⊤ : G.Subgraph).deleteVerts u).coe.oddComponents.ncard
+
+/-- A graph in which the universal vertices do not violate the
+Tutte-condition, if the graph decomposes into cliques, there exists a matching that covers
```

Comment:
I would say
```suggestion
Tutte condition, if the graph decomposes into cliques, there exists a matching that covers
```
no? Also in one other place

Reply:
Replaced with "Tutte's condition", which seems in more common use.